tools/nl/embeddings/prep/gen_svs.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import glob
+
+from absl import app
+from absl import flags
+import datacommons as dc
+import numpy as np
+import pandas as pd
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('sv_list_file', 'input_svs.csv',
+                    'File with SVs, one per line')
+flags.DEFINE_string('ddd_titles_file', '/tmp/ddd_titles.csv',
+                    'File with 101K curated titles from DDD')
+flags.DEFINE_string('existing_svs_filepattern', '../data/*_input/*.csv',
+                    'Pattern to CSVs with existing SVs')
+flags.DEFINE_string('output_file', 'autogen_svs.csv', 'Output file path')
+
+
+# Use DC API to get names for SVs.
+def get_names(dcids):
+  names = []
+  index = 0
+  batch_num = 200
+  while index < len(dcids):
+    print(f"index={index}")
+    dcids_subset = dcids[index:index + batch_num]
+    name_dict = dc.get_property_values(dcids_subset, 'name')
+
+    for d in dcids_subset:
+      if d in name_dict:
+        name_list = name_dict[d]
+        if name_list and name_list[0]:
+          names.append(name_list[0])
+        else:
+          names.append("")
+
+    index += batch_num
+  return names
+
+
+def main(_):
+  # Get the 14k dcids.
+  with open(FLAGS.sv_list_file) as f:
+    sv_list = f.read().splitlines()
+
+  # get names and create a DataFrame with dcids and names.
+  names_list = get_names(sv_list)
+
+  input_df = pd.DataFrame.from_dict({'dcid': sv_list, "Name": names_list})
+
+  # Get the 101k dataframe.
+  ddd_df = pd.read_csv(FLAGS.ddd_titles_file).fillna("")
+  cols_to_keep = ["Name", "StatVar ID", "Chart Title"]
+  ddd_df = ddd_df[cols_to_keep]
+  ddd_df.rename(columns={"StatVar ID": "dcid"}, inplace=True)
+
+  # Handle dupes (concats duplicated DDD descriptions/titles in a semi-colon delimited manner.)
+  ddd_df = ddd_df.groupby('dcid').agg(lambda x: ';'.join(set(x))).reset_index()
+
+  # Get all the existing SVs!
+  existing_dfs = []
+  for svf in sorted(glob.glob(FLAGS.existing_svs_filepattern)):
+    # Get the 1300 curated svs (Demo_US_SVs)
+    existing_dfs.append(pd.read_csv(svf).fillna(""))
+  existing_df = pd.concat(existing_dfs)
+
+  print(len(input_df))
+  print(len(ddd_df))
+  print(len(existing_df))
```

Comment:
remove or make this full print sentence?

Reply:
Removed